src/products/tests/products.e2e-spec.ts

```diff
@@ -0,0 +1,56 @@
+import * as request from 'supertest';
+import { HttpStatus, INestApplication } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { ProductEntity } from '@products/entities/product.entity';
+import { ProductModule } from '@products/product.module';
+import { Repository } from 'typeorm';
+import { createFakeProductEntityArray } from './fakes/products.fake';
+import { sharedTestingConfig } from '@main/shared/tests/utils/shared-testing.config';
+
+describe('Products', () => {
+  let app: INestApplication;
+  let repository: Repository<ProductEntity>;
+
+  beforeAll(async () => {
+    const module = await Test.createTestingModule({
+      imports: [ProductModule, ...sharedTestingConfig],
+    }).compile();
+    app = module.createNestApplication();
+    repository = module.get<Repository<ProductEntity>>(
+      getRepositoryToken(ProductEntity),
+    );
+    await app.init();
+  });
+
+  describe('GET /products', () => {
+    it('should return empty array because there are no products', async () => {
+      const response = await request(app.getHttpServer()).get('/products');
```

Comment:
If you watch this line, you see that you are repeating the 

```
app.getHttpServer()
```
 in all your tests. It's better to just declare a global variable and make use of that in all the tests to aovid falling into a DRY violation.

Reply:
Done!